module/zfs/zfs_ioctl.c

```diff
@@ -5821,6 +5822,44 @@ zfs_ioc_send_space(const char *snapname, nvlist_t *innvl, nvlist_t *outnvl)
 	return (error);
 }
 
+/*
+ * Sync the currently open TXG to disk for the specified pool.
+ * This is somewhat simular to 'zfs_sync()'.
+ * For all cases that do not result in error this iotcl will wait for
+ * the currently open TXG to commit before returning back to the caller.
+ *
+ * innvl is unused
+ *
+ * onvl is unused
+ */
+/* ARGSUSED */
+static int
+zfs_ioc_pool_sync(const char *pool, nvlist_t *innvl, nvlist_t *onvl)
+{
+	int err;
+	spa_t *spa;
+
+	ASSERT3P(innvl, ==, NULL);
+
+	/* we don't want a panicking kernel writing to storage */
+	if (unlikely(oops_in_progress))
+		return (EHOSTDOWN);
```

Comment:
This check belongs somewhere more generic - not in every ioctl that does a txg_wait_synced().  Perhaps zfsdev_ioctl().  (Although it's surprising that any syscalls are allowed when panicking.  Pretty sure that isn't allowed on illumos.)